doc/cephfs/nfs.rst

```diff
@@ -0,0 +1,157 @@
+
+EXPORT{CLIENT {}} options
+
+- ``Clients`` is client list entry
+
+EXPORT {} and EXPORT{CLIENT {}} options
+
+- ``Protocols`` are the allowed protocols. NFSv3 and v4 protocols are allowed
+  by default.  Here it is set to '4' to export over NFSv4 protocols. It is
+  preferred that clients use NFSv4.1+ to talk to NFS-Ganesha.
+
+- ``Squash`` is the kind of user ID squashing performed. Root-squashing
+  is enabled by default. So the operations attempted by the client's root user
+  are performed as if by the local "nobody" (and "nogroup") user on the
+  NFS-Ganesha server.
+
+- ``Access_Type`` is the kind of access allowed to the clients.
+
+EXPORT {FSAL {}} options for Ceph FSAL
+
+- ``Name`` of FSAL should always be "Ceph".
+
+- ``User_Id`` is the cephx user ID used by the Ceph client to connect to
+  CephFS. The cephx ID needs to be created before running NFS-Ganesha, and
+  should have the necessary authorization to access the CephFS path exported
+  by NFS-Ganesha. See:
+  `<http://docs.ceph.com/docs/master/rados/operations/user-management/>`_
+
+- ``Secret_Access_Key`` is the secret key of the cephx ID. If not set, then it
+  uses the normal search path for cephx keyring files to find a key.
+
+CEPH {} options
+
+- ``Ceph_Conf`` is the path to the Ceph configuration file. If not specified,
+  Ceph clients use '/etc/ceph/ceph.conf'.
+
+- ``umask`` is the user file-creation mask. These bits will be masked off from
+  the unix permissions on newly created inodes. It defaults to zero.
+
+Refer to
+
+- `<https://github.com/nfs-ganesha/nfs-ganesha/blob/next/src/doc/man/ganesha-export-config.rst>`_
+  for more details on EXPORT {} and CLIENT {} options
+
+- `<https://github.com/nfs-ganesha/nfs-ganesha/blob/next/src/doc/man/ganesha-ceph-config.rst>`_
+  for more details about Ceph FSAL {} options
+
+- `<https://github.com/nfs-ganesha/nfs-ganesha/blob/next/src/config_samples/export.txt>`_
+  for more config samples
+
+- `<https://github.com/nfs-ganesha/nfs-ganesha/blob/next/src/config_samples/ceph.conf>`_
+  for more Ceph FSAL related config samples
+
+Configuration for libcephfs clients
+-----------------------------------
+
+Required ceph.conf for libcephfs clients includes:
+
+* a [client] section with ``mon_host`` option set to let the clients connect
+  to the Ceph cluster's monitors, e.g., ::
+
+    [client]
+            mon host = 192.168.1.7:6789, 192.168.1.8:6789, 192.168.1.9:6789
+
+Mount using NFSv4 clients
+=========================
+
+It is preferred to mount the NFS-Ganesha exports using NFSv4.1+ protocols
+to get the benefit of sessions.
+
+Conventions for mounting NFS resources are platform-specific. The
+following conventions work on Linux and some Unix platforms:
+
+From the command line::
+
+  mount -t nfs -o nfsvers=4.1,proto=tcp <ganesha-host-name>:/ <mount-point>
+
+Current limitations
+===================
+
+- A Ganesha server daemon cannot export more than one Ceph Filesystem.
```

Comment:
O.o

really?

Reply:
Can export multiple directories, as follows:


>EXPORT
{
        Export_Id = 11;
        Path = /test1;
        Pseudo = /test1;
        Access_Type = RW;
        Protocols = 3,4;
        Squash = No_Root_Squash;
        FSAL {
                Name = CEPH;
        }
}
EXPORT
{
        Export_Id = 12;
        Path = /test2;
        Pseudo = /test2;
        Access_Type = RW;
        Protocols = 3,4;
        Squash = No_Root_Squash;
        FSAL {
                Name = CEPH;
        }
}

Reply:
@renhwztetecs ,  yes a single Ganesha daemon can export multiple directories within a Ceph file system. Have you been able to export multiple file systems?

Reply:
@ajarr 
Yes, I can export both FSAL_VFS and FASL Ceph(Version: nfs-ganesha-2.5.4)
>EXPORT
{
        Export_Id = 14;
        Path = /test4;
        Pseudo = /test4;
        Access_Type = RW;
        Protocols = 3,4;
        Squash = No_Root_Squash;
        FSAL {
                Name = CEPH;
        }
}
EXPORT
{
        Export_Id = 77;
        Path = /abc;
        Pseudo = /abc;
        Access_Type = RW;
        Protocols = 3,4;
        Squash = No_Root_Squash;
        FSAL {
                Name = VFS;
        }
}

Reply:
Let's reword it to say "FSAL_CEPH can only export one Ceph file system although multiple directories in a Ceph file system may be exported."

Reply:
Done